plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -247,7 +254,7 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         try {
             table = catalog.loadTable(session, new SchemaTableName(tableName.getSchemaName(), name.getTableName()));
         }
-        catch (TableNotFoundException e) {
+        catch (TableNotFoundException | UnknownTableTypeException e) {
```

Comment:
suppressing `UnknownTableTypeException` looks weird here.
document why this is needed.

Reply:
the reasoning behind this choice is to avoid dealing with non Iceberg tables. I've added documentation for this specific `catch` clause